include/subprocess/subprocess.hpp

```diff
@@ -102,6 +89,39 @@ class command_error : public subprocess_error
 };
 } // namespace exceptions
 
+/**
+ * @brief Raise a std::system_error when a syscall fails
+ *
+ * @param name The name of the offending function.
+ * @param errval The errno-style error code.
+ * @param extra Additional text to display in the error message.
+ */
+[[noreturn]] void raise_sys_error(std::string name, int errval, const std::string& extra = "")
```

Comment:
Can you move these functions to the exceptions namespace?

[nitpick] I don't understand the use of `extra`. Since its always known at compile time whether or not `extra` would be present, we can always use it as:

    exceptions::raise_sys_error("posix: "s + sys.argv[0])

Reply:
(I've been away from the computer for a few days). Probably moot now, but I had the formatting inside the function to make it easy to keep it consistent and avoid having to repeat basic string concatenation in multiple places.

Reply:
I was planning on adding an `operator<<` overload or a var-arg initialiser list overload to std::subprocess down the line. Let's do that. That will come in handy.